app/models/recommendation.py

```diff
@@ -98,16 +98,17 @@ async def get_recommendations_from_experiment(
 
         # apply rankers from the slate experiment on the candidate set's candidates
         for ranker in experiment.rankers:
-            if ranker == 'thompson-sampling':
+            if ranker in THOMPSON_SAMPLING_RANKERS:
                 # thompson sampling takes two specific arguments so it needs to be handled differently
-                recommendations = await RecommendationModel.__thompson_sample(slate_id, recommendations)
+                recommendations = await RecommendationModel.__thompson_sample(ranker, slate_id, recommendations)
                 continue
             recommendations = get_ranker(ranker)(recommendations)
 
         return recommendations
 
     @staticmethod
-    async def __thompson_sample(slate_id: str, recommendations: ['RecommendationModel']) -> ['RecommendationModel']:
+    async def __thompson_sample(ranker: str, slate_id: str,
```

Comment:
Nit: I like to clearly distinguish an id/name from the object that does the actual work.
```suggestion
    async def __thompson_sample(ranker_name: str, slate_id: str,
```